src/ifaddrs.rs

```diff
@@ -42,11 +42,47 @@ cfg_if! {
     }
 }
 
+/// Workaround a bug in XNU where netmasks will always have the wrong size in
+/// the sa_len field due to the kernel ignoring trailing zeroes in the future
+/// when setting the field. See https://github.com/nix-rust/nix/issues/1709#issuecomment-1199304470
+///
+/// To fix this, we overwrite sa_len to have the known (static) value of the
+/// structure.
+#[cfg(target_os = "macos")]
+fn workaround_xnu_bug(info: &libc::ifaddrs) {
+    if info.ifa_netmask.is_null() {
+        return
+    }
+    let family = unsafe {
+        (*info.ifa_netmask).sa_family
+    };
+
+    let fixup_size = match i32::from(family) {
+        libc::AF_INET => Some(std::mem::size_of::<libc::sockaddr_in>()),
+        libc::AF_INET6 => Some(std::mem::size_of::<libc::sockaddr_in6>()),
+        _ => None
+    };
+
+    if let Some(fixup_size) = fixup_size {
+        assert!(fixup_size < u8::MAX.into());
+        let sock = info.ifa_netmask as *mut libc::sockaddr;
+        unsafe {
+            if (*sock).sa_len < fixup_size as u8 {
+                (*sock).sa_len = fixup_size as u8;
+            }
+        }
+    }
```

Comment:
```suggestion
    assert!(fixup_size < u8::MAX.into());
    let sock = info.ifa_netmask as *mut libc::sockaddr;
    unsafe { (*sock).sa_len = (*sock).sa_len.max(fixup_size as u8) }
```

Also, seems like `ifa_netmask` is already of the type you're casting it to ? https://docs.rs/libc/latest/x86_64-apple-darwin/libc/struct.ifaddrs.html#structfield.ifa_netmask